### PR TITLE
Avoid implementing `var_type(v)` as `v->type & 0x7F`

### PR DESCRIPTION
**Pull request by skiars**
This macro `var_type` is used extensively. This definition may bring significant performance overhead:
``` c
#define var_type(_v) ((_v)->type & 0x7F)
```

So I tried to avoid it. These modifications have passed the tests, but for the time being, it is still uncertain whether they are fully covered (for example, `debug.attrdump`).

### CODE REVIEW

**Comment by skiars** [comment]
There is obviously something wrong with `debug.attrdump`...

File: src/be_object.h [hunk 1]
```diff
@@ -34,11 +34,7 @@
 #define BE_NTVFUNC      ((0 << 5) | BE_FUNCTION)
 #define BE_CLOSURE      ((1 << 5) | BE_FUNCTION)
 #define BE_NTVCLOS      ((2 << 5) | BE_FUNCTION)
-#define BE_FUNC_STATIC  (1 << 7)
-
-#define func_isstatic(o)       (((o)->type & BE_FUNC_STATIC) != 0)
-#define func_setstatic(o)      ((o)->type |= BE_FUNC_STATIC)
-#define func_clearstatic(o)    ((o)->type &= ~BE_FUNC_STATIC)
+#define BE_STATIC       (1 << 6)
```
Comment (s-hadinger) [inline src/be_object.h]: I think there is a collision with `BE_NTVCLOS`.

`2 << 5 == 1 << 6`

File: src/be_vm.c [hunk 2]
```diff
@@ -305,6 +305,7 @@ static int obj_attribute(bvm *vm, bvalue *o, bstring *attr, bvalue *dst)
             "the '%s' object has no attribute '%s'",
             str(be_instance_name(obj)), str(attr));
     }
+    var_clearstatic(dst);
```
Comment (s-hadinger) [inline src/be_vm.c:308]: This is smart ;)

**Review by s-hadinger — COMMENTED** [review]
I think it should be `#define BE_STATIC       (1 << 7)`

File: src/be_object.h [hunk 3]
```diff
@@ -11,11 +11,11 @@
 #include "berry.h"
 
 /* basic types, do not change value */
+#define BE_NONE        -1       /* unknown type */
```
Comment (Skiars) [inline src/be_object.h:14]: I have observed that `BE_NONE` is only temporarily computed when it needs to be used. It doesn't seem to be stored in any `bvalue`, so I changed it to -1 to save numbers.

In addition, I checked that the use of BE_STATIC will not be affected.

https://github.com/berry-lang/berry/blob/3d39a9e75c2ed04b1ce9f3b37ac7ae9c47c26c45/src/be_vm.c#L854

**Comment by s-hadinger** [comment]
There are some additional places where the `BE_STATIC` bit might interfere, fortunately they are in places called less often so not implying a performance hit.

I don't know how to amend this PR so I've included the changes in another PR #202

**Pull request closed without merging**